zerver/data_import/slack.py

```diff
@@ -823,10 +824,14 @@ def get_messages_iterator(
                     # skipping those messages is simpler.
                     continue
                 if message.get("mimetype") == "application/vnd.slack-docs":
-                    # This is a Slack "Post" which is HTML-formatted,
-                    # and we don't have a clean way to import at the
-                    # moment.  We skip them on import.
-                    continue
+                    # This is a Slack "Post" which is HTML-formatted
+                    file_url = message.get("url_private_download")
+                    response = requests.get(file_url)
+                    html_content = str(response.content, encoding="utf-8")
+                    # html2text is GPL licensed, so run it as a subprocess.
+                    text = subprocess.check_output(["html2text"], input=html_content, text=True)
+                    message["text"] = text
```

Comment:
The text should probably be prepended to indicate that it comes from a canvas, not a regular message.

Reply:
I have added the message as header in italics. Please let me know if I should make more customisations.
<img width="1467" alt="Screenshot 2023-10-21 at 12 15 22 PM" src="https://github.com/zulip/zulip/assets/61960532/a12ccb5c-1c99-4808-b884-1b3e33b2df94">

Reply:
From https://slack.com/intl/en-in/help/articles/220556107-How-to-read-Slack-data-exports#how-to-view-canvases, it seems that the canvas always has a title. It would be informative to include them. Or is it already included in the HTML content, and would be redundant?

Reply:
I had just created a canvas where I only included paragraph formatting, without any heading, I got `'title' : 'Untitled'` like this:
```
{
   "id":"F062BR33L1F",
   "created":1697745676,
   "name":"Untitled",
   "title":"Untitled",
   "mimetype":"application/vnd.slack-docs",
   "filetype":"quip",
   "pretty_type":"Canvas",
   "user":"USLACKBOT",
   "editable":true,
   "size":14,
   "mode":"quip",
   "is_public":true,
   "is_tombstoned":false,
   "public_url_shared":false,
   "date_delete":0,
   "url_private_download":"https://files.slack.com/files-pri/T0607KC9HDY-F062BR33L1F/download/canvas?t=xoxe-6007658323474-6072075960805-6072076009397-1239a1ac14412e755225f7e0ab64205b",
   "shares":[
      {
         "team":"T0607KC9HDY",
         "channel":"C0604MMJ1T7"
      }
 
```

Attaching the image of canvas for your reference:

<img width="1458" alt="Screenshot 2023-10-21 at 5 48 55 PM" src="https://github.com/zulip/zulip/assets/61960532/60996430-e6be-4044-8277-b690f06fe24b">